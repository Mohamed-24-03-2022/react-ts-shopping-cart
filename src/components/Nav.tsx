import { Link, useLocation } from 'react-router-dom';
import { AiOutlineShoppingCart, AiOutlineMenu } from 'react-icons/ai';

const Nav = () => {
  const location = useLocation();

  const handleClick: React.MouseEventHandler<SVGElement> = (e) => {
    document.querySelector('.menu')?.classList.toggle('hidden');
  };
  return (
    <nav className="w-full flex justify-between items-start sm:items-center p-4 md:p-6 bg-backImg text-blackLight font-bold text-xl ">
      <div className="logo">
        <Link to="/">
          <h1>LOGO</h1>
        </Link>
      </div>
      <div className="nav-items flex flex-col items-center gap-2">
        <AiOutlineMenu className="sm:hidden" onClick={handleClick} />
        
        <ul className="menu-list hidden sm:flex sm:flex-row justify-end gap-10 mr-20">
          <Link to="/">
            <li>Home</li>
          </Link>
          <Link to="/shop">
            <li>Shop</li>
          </Link>
          <Link to="/about">
            <li>About</li>
          </Link>
          {location.pathname !== '/shop' ? (
            <div className="flex justify-center items-center text-2xl">
              <Link to="/cart">
                <AiOutlineShoppingCart />
              </Link>
            </div>
          ) : (
            ''
          )}
        </ul>
        <div className="menu flex flex-col gap-2 hidden sm:hidden ">
          <Link to="/">Home</Link>
          <Link to="/shop">Shop</Link>
          <Link to="/about">About</Link>
          {location.pathname !== '/shop' ? (
            <div className="flex justify-center items-center text-2xl">
              <Link to="/cart">
                <AiOutlineShoppingCart />
              </Link>
            </div>
          ) : (
            ''
          )}
        </div>

      </div>
    </nav>
  );
};

export default Nav;
