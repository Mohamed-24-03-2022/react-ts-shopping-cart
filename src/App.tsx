import { useState } from 'react';
import Nav from './components/Nav';
import Main from './components/Main';
import ShopPage from './components/ShopPage';
import AboutPage from './components/AboutPage';
import Cart from './components/Cart';
import Footer from './components/Footer';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { laptop } from './components/data';

type bucketType = {
  item: laptop;
  quantity: number;
};

function App() {
  const [buckets, setBuckets] = useState<bucketType[]>([]);

  return (
    <BrowserRouter>
      <Nav />
      <Routes>
        <Route path="/" element={<Main />}></Route>
        <Route
          path="/shop"
          element={
            <ShopPage buckets={buckets} setBucketsOfParent={setBuckets} />
          }
        ></Route>
        <Route path="/about" element={<AboutPage />}></Route>
        <Route
          path="/cart"
          element={<Cart setBucketsOfParent={setBuckets} buckets={buckets} />}
        ></Route>
      </Routes>
      <Footer />
    </BrowserRouter>
  );
}

export default App;
