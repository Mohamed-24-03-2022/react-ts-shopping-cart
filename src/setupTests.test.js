
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';

import App from './App';
import Nav from './components/Nav';

describe('Main component', () => {

  test('renders App correctly', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  // test('renders Nav correctly', () => {
  //   const { container } = render(<Nav />);
  //   expect(container).toMatchSnapshot();
  // });

});