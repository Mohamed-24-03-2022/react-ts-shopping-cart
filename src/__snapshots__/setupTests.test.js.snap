// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders App correctly 1`] = `
<div>
  <nav
    class="w-full flex justify-between items-start sm:items-center p-4 md:p-6 bg-backImg text-blackLight font-bold text-xl "
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <h1>
          LOGO
        </h1>
      </a>
    </div>
    <div
      class="nav-items flex flex-col items-center gap-2"
    >
      <svg
        class="sm:hidden"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
        />
      </svg>
      <ul
        class="menu-list hidden sm:flex sm:flex-row justify-end gap-10 mr-20"
      >
        <a
          href="/"
        >
          <li>
            Home
          </li>
        </a>
        <a
          href="/shop"
        >
          <li>
            Shop
          </li>
        </a>
        <a
          href="/about"
        >
          <li>
            About
          </li>
        </a>
        <div
          class="flex justify-center items-center text-2xl"
        >
          <a
            href="/cart"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
              />
            </svg>
          </a>
        </div>
      </ul>
      <div
        class="menu flex flex-col gap-2 hidden sm:hidden "
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/shop"
        >
          Shop
        </a>
        <a
          href="/about"
        >
          About
        </a>
        <div
          class="flex justify-center items-center text-2xl"
        >
          <a
            href="/cart"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="flex flex-col"
  >
    <div
      class="main-container flex flex-col h-full sm:h-[600px] sm:flex-row"
    >
      <div
        class="img-container w-full h-full md:w-3/4"
      >
        <img
          alt="keyboard-img"
          class="w-full h-full"
          src="keyboard-img.jpg"
        />
      </div>
      <div
        class="hero-section flex flex-col justify-center items-start min-h-max w-full space-y-8 p-8 bg-backHero text-whiteLight "
      >
        <h1
          class="text-4xl w-full sm:w-3/4 lg:w-5/12 leading-tight font-extrabold"
        >
          Gaming Laptops Services
        </h1>
        <p
          class="w-full text-lg lg:w-3/5"
        >
          Suspendisse potenti. Donec sed mi lorem. Quisque euismod, tortor non commodo malesuada, ante nibh posuere mauris, vitae bibendum augue ipsum sed tortor. Quisque sit amet est ac turpis malesuada fermentum.
        </p>
        <button
          class="bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center"
        >
          Shop Page
        </button>
      </div>
    </div>
    <div
      class="content-container text-center p-10 text-blackLight space-y-20 bg-backImg"
    >
      <div
        class="flex flex-col  items-center gap-6"
      >
        <h1
          class="text-4xl font-bold"
        >
          Our Services
        </h1>
        <hr
          class="p-1 w-36 bg-blackLight rounded-md"
        />
        <p
          class="sm:text-left text-center w-3/4"
        >
          Seitan man braid microdosing roof party wolf semiotics lomo lorem authentic health goth pickled heirloom irony four dollar toast shoreditch. Vice knausgaard echo park microdosing jianbing asymmetrical speed test toast noo.
        </p>
      </div>
      <div
        class="card-container grid gap-20 justify-items-center justify-center"
      >
        <div
          class="space-y-8 min-w-[220px] bg-white px-6 py-16 shadow-2xl rounded-lg"
        >
          <h3
            class="font-bold text-2xl"
          >
            Laptop Maintenance
          </h3>
          <p>
            Narw tbh flannel. Hot lorem gochujangtogen.
          </p>
        </div>
        <div
          class="space-y-8 min-w-[220px] bg-white px-6 py-16 shadow-2xl rounded-lg"
        >
          <h3
            class="font-bold text-2xl"
          >
            Breakthrough Tech
          </h3>
          <p>
            Narw tbh flannel. Hot lorem gochujangtogen.
          </p>
        </div>
        <div
          class="space-y-8 min-w-[220px] bg-white px-6 py-16 shadow-2xl rounded-lg"
        >
          <h3
            class="font-bold text-2xl"
          >
            Aesthetic Design
          </h3>
          <p>
            Narw tbh flannel. Hot lorem gochujangtogen.
          </p>
        </div>
      </div>
      <div>
        <button
          class="bg-white hover:bg-gray-100 text-gray-800 font-semibold py-2 px-4 border border-gray-400 rounded shadow"
        >
          Show All Services
        </button>
      </div>
    </div>
    <div
      class="testimonies content-container p-10 space-y-20 bg-backHero"
    >
      <div
        class="flex flex-col items-center gap-4 text-whiteLight"
      >
        <h1
          class="text-4xl sm:text-6xl font-bold min-w-max"
        >
          What Clients Say
        </h1>
        <hr
          class="pt-1 sm:p-1 w-3/5 sm:w-1/5 bg-whiteLight rounded-md"
        />
      </div>
      <div
        class="card-container grid gap-20 justify-center text-blackLight"
      >
        <div
          class="space-y-8 px-6 py-12 sm:py-24 shadow-2xl rounded-lg bg-white"
        >
          <p>
            Mustache flexitarian prism sriracha enamel pin offal meh shabby chic four loko meggings subway tile portland pickled narwhal ennui truffaut neutra shaman.
          </p>
        </div>
        <div
          class="space-y-8 px-6 py-12 sm:py-24 shadow-2xl rounded-lg bg-white"
        >
          <p>
            Mustache flexitarian prism sriracha enamel pin offal meh shabby chic four loko meggings subway tile portland pickled narwhal ennui truffaut neutra shaman.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex justify-center items-center p-4 bg-backImg text-xs sm:text-base"
  >
    <p>
      copyright Â© 2023 Mohamed-24-03-2022
    </p>
    <a
      class="mx-1"
      href="https://github.com/Mohamed-24-03-2022"
      rel="author"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
        />
      </svg>
    </a>
  </div>
</div>
`;
